pip/req/req_install.py

```diff
@@ -265,6 +266,15 @@ def link(self, link):
     def specifier(self):
         return self.req.specifier
 
+    @property
+    def is_pinned(self):
+        """Return whether I am pinned to an exact version.
+
+        For example, some-package==1.2 is pinned; some-package>1.2 is not.
+        """
+        specifiers = self.specifier
+        return len(specifiers) == 1 and next(iter(specifiers)).operator == '=='
```

Comment:
It's a little bit odd here, because with PEP 440 ``==`` doesn't actually guarantee you'll *always* get the exact same thing. Local versions are not considered when checking ``==`` so if someone uploads ``1.0+mylocalversion`` that will match ``==1.0``. However PyPI does not allow uploads of local versions, so it might be a moot point.

The ``===`` is a way to essentially escape PEP 440's strict parsing requirements for legacy versions that can't be successfully parsed as a PEP 440 version. It also provides a way to escape the ``==`` behavior around local versions.